app/static/src/app/store/appState/actions.ts

```diff
@@ -40,6 +41,7 @@ export const appStateActions: ActionTree<AppState, AppState> = {
 
         if (response) {
             commit(`code/${CodeMutation.SetCurrentCode}`, state.config!.defaultCode, { root: true });
+            commit(`run/${RunMutation.SetDefaultEndTime}`, response.data.endTime, {root: true});
```

Comment:
A side effect of the approach I've take is that the endTime is being set in two places in the store: in run where it is used and in the appState config where it doesn't do anything. I could go out of my way to ensure it isn't saved in the second place if you think that is required?